bin/render-activity-stream-html.js

```diff
@@ -69,54 +85,51 @@ function getTextDirection(locale) {
  */
 function templateHTML(options, html) {
   const isPrerendered = !!html;
-  const debugString = options.debug ? "-dev" : "";
-  const scripts = [
-    "chrome://browser/content/contentSearchUI.js",
-    "chrome://browser/content/contentTheme.js",
-    `${options.baseUrl}vendor/react${debugString}.js`,
-    `${options.baseUrl}vendor/react-dom${debugString}.js`,
-    `${options.baseUrl}vendor/prop-types.js`,
-    `${options.baseUrl}vendor/react-intl.js`,
-    `${options.baseUrl}vendor/redux.js`,
-    `${options.baseUrl}vendor/react-redux.js`,
-    `${options.baseUrl}prerendered/${options.locale}/activity-stream-strings.js`,
-    `${options.baseUrl}data/content/activity-stream.bundle.js`,
-  ];
+  const scripts = getScripts(options);
   if (isPrerendered) {
     scripts.unshift(`${options.baseUrl}prerendered/static/activity-stream-initial-state.js`);
   }
-  const scriptTag = `
-    <script>
-// Don't directly load the following scripts as part of html to let the page
-// finish loading to render the content sooner.
-for (const src of ${JSON.stringify(scripts, null, 2)}) {
-  // These dynamically inserted scripts by default are async, but we need them
-  // to load in the desired order (i.e., bundle last).
-  const script = document.body.appendChild(document.createElement("script"));
-  script.async = false;
-  script.src = src;
-}
-    </script>`;
+  const loadScriptTag = `<script src="${options.baseUrl}prerendered/static/activity-stream-load-scripts.js"></script>`;
+  const renderInfo = [
+    isPrerendered && "prerendered",
+    options.noscripts ? "noscripts" : "scripts-loaded",
+    options.debug && "debug",
+  ].filter(i => i);
   return `<!doctype html>
 <html lang="${options.locale}" dir="${options.direction}">
+  <!-- Render info: ${renderInfo.join(", ")} -->
   <head>
     <meta charset="utf-8">
-    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'unsafe-inline' resource: chrome:; connect-src https:; img-src https: data: blob:; style-src 'unsafe-inline';">
+    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; object-src 'none'; script-src resource: chrome:; connect-src https:; img-src https: data: blob:; style-src 'unsafe-inline';">
     <title>${options.strings.newtab_page_title}</title>
     <link rel="icon" type="image/png" href="chrome://branding/content/icon32.png"/>
     <link rel="stylesheet" href="chrome://browser/content/contentSearchUI.css" />
     <link rel="stylesheet" href="${options.baseUrl}css/activity-stream.css" />
   </head>
   <body class="activity-stream">
-    <div id="root">${isPrerendered ? html : "<!-- Regular React Rendering -->"}</div>
+    <div id="root"></div>
```

Comment:
Nothing will get prerendered without the `html` (unless I'm missing where it gets prerendered elsewhere??)

Reply:
Whoops sorry, the `html` should still be in there